go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -25,18 +26,53 @@ const (
 	defaultOpenFlowCookie = "0xdeff105"
 )
 
-// nodePortWatcher manages OpenfLow and iptables rules
+// nodePortWatcher manages Openflow and iptables rules
 // to ensure that services using NodePorts are accessible
 type nodePortWatcher struct {
+	nodeName    string
+	nodeIPs     []string
 	ofportPhys  string
 	ofportPatch string
 	gwBridge    string
-	ofm         *openflowManager
+	// The services Map is required to decide if we care about a given Endpoint event
+	services map[ktypes.NamespacedName]*kapi.Service
+	ofm      *openflowManager
 }
 
-func (npw *nodePortWatcher) updateServiceFlowCache(service *kapi.Service, add bool) {
+// With The external Traffic policy feature this logic got much more complicated
+// Now we have Multiple K8s Object tiggers (i.e Services and Endpoints) That must
+// Prompt a single event function -->
+// func updateServiceFlowCache(service *kapi.Service, add bool, epLocal bool)
+//
+// To handle this, two signal variables are used, add and epLocal
+// If add==false all br-ex flows are deleted for the service
+// If add==true br-ex flows are created for the service +
+// 		if epLocal==False && ETP==Local it means the svc has no Host Endpoints
+//         so we add only a single flow to on br-ex to steer traffic into OVN-K
+//
+//      if epLocal==True && ETP==Local it means the svc has Host endpoints
+//         so we add 4 flows onto br-ex to allow external-> SVC traffic to
+//         Completely bypass OVN.
+//
+// For Service Event functions epLocal is always FALSE since we're looking only at the service
+// and know nothing about it's endpoints
+//
+// For endpoint Event functions...
+//		An action only ever occurs (i.e epLocal=True) if
+//			1. The Service Exists &&
+// 			2. Svc's ETP==Local &&
+//			3. Svc Has host networked Backends
+//      Otherwise the endpoint event is ignored
+//
+//      The action ensures the correct flows and IPtables rules are programmed
+
+// This function is used to manage the flows programmed in br-ex for OVN-K8's Nodeport, ExternalIP, and
+// Loadbalancer type services. It also consumes an `epLocal` argument to signal that the specified service
+// has has host-networked backends
```

Comment:
This is super confusing - HostNetwork has a very specific meaning - namely, that a pod share's the host's network namespace (and thus has the same IP).

If I'm reading this correctly, you seem to be using it to mean that the pod is hosted locally on the node, but it has its own network namespace.

Am I reading this correctly? Can you clarify?

Reply:
Correct